src/lib/RequestParser.js

```diff
@@ -256,6 +256,22 @@ class RequestParser { // eslint-disable-line no-unused-vars
         return !!value;
     }
 
+    /**
+     * Checks that a given value is a non-negative finite number.
+     * @param {any} value
+     * @param {boolean} [allowUndefined=true]
+     * @returns {number | undefined}
+     */
+    parseNonNegativeFiniteNumber(value, allowUndefined = true) {
+        if (value === undefined && allowUndefined) {
```

Comment:
I would swap the conditions
```suggestion
        if (allowUndefined && value === undefined) {
```
as the second condition is only relevant iff the first one is fulfilled.

Reply:
The other way around the second one (`allowUndefined`) is also only relevant when the first one is fulfilled :wink:

Reply:
IMO it is better as it is.

Reply:
I like it better the other way around, beginning with the boolean that toggles this functionality.

Reply:
I think this doesn't really matter, but I also prefer how it is currently.